src/common/PropertyExt.js

```diff
@@ -0,0 +1,254 @@
+    } else {
+        root.common_PropertyExt = factory(root.d3);
+    }
+}(this, function (d3) {
+    var __meta_ = "__meta_";
+    var __prop_ = "__prop_";
+
+    function Meta(id, defaultValue, type, description, set, ext) {
+        this.id = id;
+        this.type = type;
+        this.origDefaultValue = defaultValue;
+        this.defaultValue = defaultValue;
+        this.description = description;
+        this.set = set;
+        this.ext = ext || {};
+
+        switch (type) {
+            case "set":
+                this.checkedAssign = function (_) {
+                    if (!set || set.indexOf(_) < 0) {
+                        console.log("Invalid value for '" + id + "':  " + _ + " expected " + type);
+                    }
+                    return _;
+                };
+                break;
+            case "html-color":
+                this.checkedAssign = function (_) {
+                    if (window.__hpcc_debug && _ && _ !== "red") {
+                        var litmus = "red";
+                        var d = document.createElement("div");
+                        d.style.color = litmus;
+                        d.style.color = _;
+                        //Element's style.color will be reverted to litmus or set to "" if an invalid color is given
+                        if (d.style.color === litmus || d.style.color === "") {
+                            console.log("Invalid value for '" + id + "':  " + _ + " expected " + type);
+                        }
+                    }
+                    return _;
+                };
+                break;
+            case "boolean":
+                this.checkedAssign = function (_) {
+                    return typeof (_) === "string" && ["false", "off", "0"].indexOf(_.toLowerCase()) >= 0 ? false : Boolean(_);
+                };
+                break;
+            case "number":
+                this.checkedAssign = function (_) {
+                    return Number(_);
+                };
+                break;
+            case "string":
+                this.checkedAssign = function (_) {
+                    return String(_);
+                };
+                break;
+            case "array":
+                this.checkedAssign = function (_) {
+                    if (!(_ instanceof Array)) {
+                        console.log("Invalid value for '" + id + "':  " + _ + " expected " + type);
+                    }
+                    return _;
+                };
+                break;
+            case "object":
+                this.checkedAssign = function (_) {
+                    if (!(_ instanceof Object)) {
+                        console.log("Invalid value for '" + id + "':  " + _ + " expected " + type);
+                    }
+                    return _;
+                };
+                break;
+            case "widget":
+            case "widgetArray":
+                this.checkedAssign = function (_) {
+                    //  TODO  ---
```

Comment:
The TODO is for me to add a test that _ === WidgetArray (I don't have a clean way yet)